third_party/googletest/src/googletest/src/gtest.cc

```diff
@@ -3989,20 +4141,29 @@ XmlUnitTestResultPrinter::XmlUnitTestResultPrinter(const char* output_file)
 // Called after the unit test ends.
 void XmlUnitTestResultPrinter::OnTestIterationEnd(const UnitTest& unit_test,
                                                   int /*iteration*/) {
+#if GTEST_OS_STARBOARD
+  std::stringstream stream;
+  PrintXmlUnitTest(&stream, unit_test);
+
+  WriteOuputFile(output_file_, StringStreamToString(&stream));
+#else
   FILE* xmlout = OpenFileForWriting(output_file_);
   std::stringstream stream;
   PrintXmlUnitTest(&stream, unit_test);
   fprintf(xmlout, "%s", StringStreamToString(&stream).c_str());
   fclose(xmlout);
+#endif
 }
 
 void XmlUnitTestResultPrinter::ListTestsMatchingFilter(
     const std::vector<TestSuite*>& test_suites) {
+#if !GTEST_OS_STARBOARD
```

Comment:
This seems like pretty important functionality, we should have at least a follow-up TODO to enable it later